src/DocumentDB.ChangeFeedProcessor/Exceptions/LeaseLostException.cs

```diff
@@ -8,46 +8,86 @@ namespace Microsoft.Azure.Documents.ChangeFeedProcessor.Exceptions
     using System.Runtime.Serialization;
     using Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement;
 
+    /// <summary>
+    /// Exception occurred when lease was taken by a different host.
+    /// </summary>
     [Serializable]
-    internal class LeaseLostException : Exception
+    public class LeaseLostException : Exception
     {
-        /// <summary>Initializes a new instance of the <see cref="LeaseLostException" /> class using default values.</summary>
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LeaseLostException" /> class using default values.
+        /// </summary>
         public LeaseLostException()
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LeaseLostException" /> class using default values.
+        /// </summary>
+        /// <param name="lease">Instance of a lease that was taken by a different host.</param>
         public LeaseLostException(ILease lease)
         {
             this.Lease = lease;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LeaseLostException" /> class using default values.
+        /// </summary>
+        /// <param name="lease">Instance of a lease that was taken by a different host.</param>
+        /// <param name="innerException">The inner exception.</param>
+        /// <param name="isGone">Whether lease doesn't exist.</param>
         public LeaseLostException(ILease lease, Exception innerException, bool isGone = false)
             : base(null, innerException)
         {
             this.Lease = lease;
             this.IsGone = isGone;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LeaseLostException" /> class using default values.
+        /// </summary>
+        /// <param name="message">The exception error message.</param>
         public LeaseLostException(string message)
```

Comment:
If we are making the exception public for catching, can't we leave the constructors internal or do we want users to be able to create them too?

Reply:
We'd like to make them public for throwing as well.